### PR TITLE
feat(dpe-api-oai): Resolve contributor information in OAI-PMH output (DEV-6575)

### PR DESCRIPTION
**Pull request by siers**
## Personal Acceptance Criteria

* In the ListRecords request in OAI, the contributors seem to have the right information
* An edge case is mentioned in the Linear issue, but doesn't seem to be applicable
* The PR mostly consists of ResolvedAgent lookups, XML changes, DI and golden test updates, seems correct

## Motivation
OAI-PMH metadata exposed contributors, creators, and funders as internal IDs (`person-028`, `0803-organization-000`), which is useless for harvesters. Person and organization data (names, ORCID/GND identifiers, affiliations) already exists in the dataset and is shown by the DPE web UI, but the OAI module never resolved it.

## Summary
- DataCite creators/contributors now carry the resolved name ("Family, Given"), `givenName`/`familyName`, ORCID/GND `nameIdentifier`, and `affiliation` elements; organizations get `nameType="Organizational"`
- DataCite `funderName` and Dublin Core `dc:creator`/`dc:contributor` are resolved to display names
- Unresolvable IDs fall back to the raw ID so no attribution is lost

### dpe-core
- New `ContributorLookup` trait with a production `CachedContributorLookup` backed by the existing person/organization caches
- Person-vs-organization ID heuristic extracted to `is_organization_id()` and shared with dpe-web (previously duplicated)

### dpe-api-oai
- New `metadata/resolve.rs`: resolves an ID to display name, nameType, given/family names, ORCID/GND name identifiers, and affiliation organization names
- `project_to_datacite` / `project_to_dublin_core` take a `&dyn ContributorLookup`, threaded through the verb handlers like the existing repository parameters
- XML writer emits `givenName`, `familyName`, `nameIdentifier`, `affiliation` in DataCite kernel-4 element order

### Testability of the disk-backed caches
**Problem:** person/organization caches are global `OnceLock`s loaded from the data directory; transforms calling them directly would be untestable.
**Solution:** repository-style trait injection (`&dyn ContributorLookup`), matching how handlers already receive `ProjectRepository`/`RecordRepository`. Tests use an in-memory lookup with fixtures.

### What identifier data exists
- Person `sameAs` carries ORCID (96), URL (58), GND (5). `URL` entries are deliberately not emitted as `nameIdentifier` (not an identifier scheme).
- No organization has a ROR `sameAs` entry yet, so `affiliationIdentifier` and `funderIdentifier` are not emitted. Add once the data exists.

### Creator detection left unchanged (deliberate)
38 of 83 projects have no attribution matching the `is_creator` rules (free-text role vocabulary: "Applicant", "PI", "Projektleitung", compound strings like "Project Leader, Data Collector") and fall back to `creatorName=DaSCH`. Out of scope here; documented separately.

## Gotchas
- Records (vs. projects) use free-text `legal_info.authorship` names, not IDs — they need no resolution and are untouched.
- Golden files were regenerated; all responses still validate against the DataCite kernel-4 XSD via xmllint.

## Test Plan
- 8 unit tests for the resolver (name formatting, ORCID/GND mapping, affiliation resolution, org contributors, raw-ID fallbacks)
- Golden tests now cover a personal creator, personal contributor with ORCID + affiliation, organizational contributor, and resolved funder
- `just check` and `just test` pass (232 tests)

### CODE REVIEW

File: modules/dpe/core/src/contributors.rs [hunk 1]
```diff
@@ -21,6 +21,37 @@ pub enum ResolvedContributor {
     },
 }
 
+/// Heuristic for distinguishing organization IDs (e.g. `organization-001`,
+/// `0803-organization-000`) from person IDs (e.g. `person-028`).
+pub fn is_organization_id(id: &str) -> bool {
+    id.starts_with("organization-") || id.contains("-organization-")
```
Comment (BalduinLandolt) [inline modules/dpe/core/src/contributors.rs]: I don't think we should have the latter case in data anymore

**Review by BalduinLandolt — APPROVED** [review]

File: modules/dpe/core/src/contributors.rs (hunk 1 above)
Comment (siers) [reply modules/dpe/core/src/contributors.rs]: Confirmed — no `-organization-` infix IDs in the data. Removed the dead branch in 7995e775.
